ray-curator/ray_curator/utils/performance_utils.py

```diff
@@ -0,0 +1,148 @@
+"""Performance utilities for tracking stage performance metrics."""
+
+from __future__ import annotations
+
+import contextlib
+import statistics
+import time
+from typing import TYPE_CHECKING
+
+import attrs
+from loguru import logger
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+
+    from ray_curator.stages.base import ProcessingStage
+
+
+@attrs.define
+class StagePerfStats:
```

Comment:
@abhinavg4 in a followup PR let's move this to dataclass for consistency across the codebase